sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_lease_client.hpp

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <chrono>
+
+#include "azure/storage/files/shares/share_client.hpp"
+#include "azure/storage/files/shares/share_file_client.hpp"
+
+namespace Azure { namespace Storage { namespace Files { namespace Shares {
+
+  /**
+   * @brief ShareLeaseClient allows you to manipulate Azure Storage leases on shares and files.
+   */
+  class ShareLeaseClient {
+  public:
+    /**
+     * @brief Initializes a new instance of the ShareLeaseClient.
+     *
+     * @param fileClient A ShareFileClient representing the file being leased.
+     * @param leaseId A lease ID. This is not required for break operation.
+     */
+    explicit ShareLeaseClient(ShareFileClient fileClient, std::string leaseId)
+        : m_fileClient(std::move(fileClient)), m_leaseId(std::move(leaseId))
+    {
+    }
+
+    /**
+     * @brief Initializes a new instance of the ShareLeaseClient.
+     *
+     * @param shareClient A ShareClient representing the share being leased.
+     * @param leaseId A lease ID. This is not required for break operation.
+     */
+    explicit ShareLeaseClient(ShareClient shareClient, std::string leaseId)
```

Comment:
Does share support lease operations? https://docs.microsoft.com/en-us/rest/api/storageservices/operations-on-shares--file-service-

Reply:
Yes, take a look at what .NET does: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Files.Shares/src/ShareLeaseClient.cs#L34

Reply:
If you cover it with unit test cases and it actually works, then i'm fine even if it's not documented.

Reply:
Seems the share lease still not ready on service end. Moved it to private just as .NET.

Reply:
Do you know in which STG will this feature be available?

Reply:
If it won't be available in the foreseeable future, I tend to delete them, including constructor, renew and test cases.

Reply:
It was said to be moved to STG75, I assume it is not turned on yet.